internal/operands/tekton-tasks/tekton-tasks.go

```diff
@@ -0,0 +1,313 @@
+		o := ct.DeepCopy()
+		objects = append(objects, o)
+	}
+
+	return common.DeleteAll(request, objects...)
+}
+
+func listDeprecatedClusterTasks(request *common.Request) ([]pipeline.ClusterTask, error) {
+	deprecatedClusterTasks := &pipeline.ClusterTaskList{}
+	err := request.Client.List(request.Context, deprecatedClusterTasks, &client.MatchingLabels{
+		common.AppKubernetesManagedByLabel: common.AppKubernetesManagedByValue,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return deprecatedClusterTasks.Items, nil
+
+}
+func isUpgradingNow(request *common.Request) bool {
+	return request.Instance.Status.ObservedVersion != environment.GetOperatorVersion()
+}
+
+func reconcileTektonTasksFuncs(tasks []pipeline.Task) []common.ReconcileFunc {
+	funcs := make([]common.ReconcileFunc, 0, len(tasks))
+	for i := range tasks {
+		task := &tasks[i]
+		funcs = append(funcs, func(request *common.Request) (common.ReconcileResult, error) {
+			task.Namespace = request.Instance.Namespace
+
+			if request.Instance.Spec.Tasks != nil && request.Instance.Spec.Tasks.Namespace != "" {
+				task.Namespace = request.Instance.Spec.Tasks.Namespace
+			}
+
+			if task.Name == modifyWindowsVMIsoFileName {
+				for i, step := range task.Spec.Steps {
+					if step.Name == "create-iso-file" {
+						task.Spec.Steps[i].Image = AllowedTasks[modifyDataObjectTaskName]()
+					}
+					if step.Name == "convert-iso-file" || step.Name == "modify-iso-file" {
+						task.Spec.Steps[i].Image = AllowedTasks[diskVirtCustomizeTaskName]()
+					}
+				}
+			} else {
+				task.Spec.Steps[0].Image = AllowedTasks[task.Name]()
+			}
+			task.Labels[TektonTasksVersionLabel] = operands.TektonTasksVersion
+			return common.CreateOrUpdate(request).
+				ClusterResource(task).
+				WithAppLabels(operandName, operandComponent).
+				UpdateFunc(func(newRes, foundRes client.Object) {
+					newTask := newRes.(*pipeline.Task)
+					foundTask := foundRes.(*pipeline.Task)
+					foundTask.Spec = newTask.Spec
+				}).
+				Reconcile()
+		})
+	}
+	return funcs
+}
+
+func reconcileClusterRoleFuncs(crs []rbac.ClusterRole) []common.ReconcileFunc {
+	funcs := make([]common.ReconcileFunc, 0, len(crs))
+	for i := range crs {
+		cr := &crs[i]
+		funcs = append(funcs, func(request *common.Request) (common.ReconcileResult, error) {
+			return common.CreateOrUpdate(request).
+				ClusterResource(cr).
+				WithAppLabels(operandName, operandComponent).
+				UpdateFunc(func(newRes, foundRes client.Object) {
+					newTask := newRes.(*rbac.ClusterRole)
+					foundTask := foundRes.(*rbac.ClusterRole)
+					foundTask.Rules = newTask.Rules
+				}).
+				Reconcile()
+		})
+	}
+	return funcs
+}
+
+func reconcileServiceAccountsFuncs(sas []v1.ServiceAccount) []common.ReconcileFunc {
+	funcs := make([]common.ReconcileFunc, 0, len(sas))
+	for i := range sas {
+		sa := &sas[i]
+		funcs = append(funcs, func(request *common.Request) (common.ReconcileResult, error) {
+			namespace := request.Instance.Namespace
```

Comment:
This is incorrect namespace.

Reply:
It should be `namespace := request.Instance.Spec.TektonPipelines.Namespace`, right?

Reply:
Okay, I'll do something similar as was done in another function:
```
namespace = request.Instance.Namespace
if request.Instance.Spec.TektonTasks != nil && request.Instance.Spec.TektonTasks.Namespace != "" {
		namespace = request.Instance.Spec.TektonTasks.Namespace
}
```

Reply:
Done.

Reply:
Please do this change.